packages/nexrender-core/src/tasks/script.js

```diff
@@ -32,32 +32,11 @@ const wrapFootage = ({ dest, ...asset }) => (`(function() {
 
 const wrapData = ({ property, value, expression, ...asset }) => (`(function() {
     ${selectLayers(asset, /* syntax:js */`function(layer) {
-        var parts = ${JSON.stringify(partsOfKeypath(property))};
-
-        var processAttribute = false;
-        var iterator = layer;
-        for (var i = 0; i < parts.length; i++) {
-            var part = parts[i];
-            if ("property" in iterator && iterator.property(part)) {
-                iterator = iterator.property(part);     
-            } else if (part in iterator) {
-                if (i + 1 < parts.length) {
-                    iterator = iterator[part]
-                } else {
-                    processAttribute = true;
-                }
-            } else {
-                throw new Error("nexrender: Can't find a property sequence (${property}) at part: " + part);                
-            }
-        }
 
-        if (processAttribute) {
-            ${renderIf(value, `iterator[parts[parts.length - 1]] = $value;`)}
-            ${renderIf(expression, `iterator[parts[parts.length - 1]] = eval($value);`)}
-        } else {
-            ${renderIf(value, `iterator.setValue($value);`)}
-            ${renderIf(expression, `iterator.expression = $value;`)}
-        }
+        var parts = ${JSON.stringify(partsOfKeypath(property))};
+        ${renderIf(value, `var value = { "value": $value }`)}
+        ${renderIf(expression, `var value = { "expression": $value }`)}
+        nexrender.changeValueForKeypath(layer, parts, value);
```

Comment:
I guess there are cases when users can use both expression and value at the same time
for example as a starting value and evolving expression that is set to work after a specific time
would it be possible to update it so it would account for that, how do you think?

Reply:
So you mean something like this for a single asset?

```
  value: "[someStartValue]"
  expression: "[someExpressionWorkingAfterSpecificTime]"
```

To speculate among AE-non-experts, my strong suspicion is that in AE you can set only *either* value *or* expression. When you set one any preceding setting will be ignored.
 But what I think you mean by "a starting value and evolving expression that is set to work after a specific time" is naturally done by using expressions only. This is because `someExpressionWorkingAfterSpecificTime` will most probably be of the form 

```
[someExpressionChangingOverTime] > [specificTime] ?  [evolve]  : [else]
```

And so, you can simply use `startValue` in the `else` part.

But thank you for the question! Thinking about it made me finally get a hunch what the distinction between property and JS attribute in AE is about. Everything that the AE team deemed "evolvable over time" is a property and the expression mechanism takes care of the evolve part. Everything else is a normal JS attribute and is set only once.

Reply:
Oh, yea, actually you are right, it should be either one or another.

Could you please update the readme, to reflect these changes:
https://github.com/inlife/nexrender#example-1

And also modify check in the validator, to allow only single item selected:
https://github.com/inlife/nexrender/blob/master/packages/nexrender-types/job.js#L65

Reply:
> Could you please update the readme, to reflect these changes in the last part of the example:

Yes, after the nudge of JamesBotterill on Issue #190, I have already added extended this example in this pull request like this:

```
{
    "assets": [
        {
            "type": "data",
            "layerName": "MyNicePicture.jpg",
            "property": "Position",
            "value": [500, 100]
        },
        {
            "type": "data",
            "layerName": "my text field",
            "property": "Source Text",
            "expression": "time > 100 ? 'Bye bye' : 'Hello world'"
        },
        {
            "type": "data",
            "layerName": "my text field",
            "property": "Source Text.font",
            "value": "Arial-BoldItalicMT"
        },
        {
            "type": "data",
            "layerName": "background",
            "property": "Effects.Skin_Color.Color",
            "value": [1, 0, 0]
        },
        {
            "type": "data",
            "layerIndex": 15,
            "property": "Scale",
            "value": [0,0],
            "expression": "[time * 0.1, time * 0.1]"
        },
    ]
}
```

The third entry, inspired by franeko #201, is new and only possible with this new feature. I have also added a sentence to the detailed paragraph about properties above the example.

> And also modify check in the validator, to allow only single item selected:

Ok, I will have a look.

Reply:
Sorry, I didn't specify properly.
I meant the last part in the last example asset:
```js
            "value": [0,0],
            "expression": "[time * 0.1, time * 0.1]"
```

Thank you @brassel!

Reply:
> I meant the last part in the last example asset:

Ah right. Done!

Regarding the other part of changing the validator, I have taken a look. If we implement the change by assert it will break backward compatibility. Some projects (including ours maybe) might stop working if people did sloppily use both value and expression. Is there a way to issue a warning from that part of  `job.js` instead of an error?

Reply:
well, unfortunately since there is no access to settings and logger we can't really log it out
(unless we just use console.log, which i dont think is a great idea)

i'm going to try to remember this spot, so later we could somehow deprecate it, or at least notify users

but overall i think it is great